script.js

```diff
@@ -0,0 +1,32 @@
+const collapeConsecutiveValues = (str = '') => {
+	if (str.length === 0) return ''
+
+	str = str
+		.split('')
+		.filter((item) => {
+			return /[a-zA-Z]/.test(item)
+		})
+		.join('')
+
+	return str
+		.match(/([a-zA-Z])\1*/g)
+		.map((item) => {
+			if (item.length === 1) return item
+			return item[0] + item.length
+		})
+		.join('')
+}
+
+// test cases:
+// 1. `aaabccccdd` → `a3bc4d2`
```

Comment:
Nice use of extended characters. It adds clarity and doubles as canary if any UTF-8 encoding issues crop up.